src/librbd/ImageWatcher.h

```diff
@@ -58,62 +58,72 @@ class ImageWatcher {
   static void notify_header_update(librados::IoCtx &io_ctx,
                                    const std::string &oid);
 
-  uint64_t get_watch_handle() const {
-    RWLock::RLocker watch_locker(m_watch_lock);
-    return m_watch_handle;
-  }
+  bool handle_payload(const watch_notify::HeaderUpdatePayload& payload,
```

Comment:
Once the ```HandlePayloadVisitor``` is moved to "librbd/watcher/Types.h" as a new templated class, ```ImageWatcher``` can just befriend the class to allow access to these